src/audio/google_rtc_audio_processing_mock.c

```diff
@@ -22,29 +24,76 @@ struct GoogleRtcAudioProcessingState {
 	int16_t *aec_reference;
 };
 
-GoogleRtcAudioProcessingState *GoogleRtcAudioProcessingCreate()
+static void SetFormats(GoogleRtcAudioProcessingState *const state,
+		       capture_sample_rate_hz,
+		       int num_capture_input_channels,
+		       int num_capture_output_channels,
+		       int render_sample_rate_hz,
+		       int num_render_channels)
+{
+	state->num_capture_channels = num_capture_input_channels;
+	state->num_output_channels = num_capture_output_channels;
+	state->num_frames = capture_sample_rate_hz / GOOGLE_RTC_AUDIO_PROCESSING_FRAMES_PER_SECOND;
+
+	state->num_aec_reference_channels = num_render_channels;
+	if (state->aec_reference) {
+		rfree(state->aec_reference);
```

Comment:
`if` isn't needed

Reply:
I think it is needed, since SetFormats could both be called with state->aec_reference allocated, and being NULL.

E.g., on line 72 it is NULL, and on line 112 it is allocated.